moveit_configs_utils/moveit_configs_utils/moveit_configs_builder.py

```diff
@@ -0,0 +1,303 @@
+        return parameters
+
+
+class MoveItConfigsBuilder(ParameterBuilder):
+    __moveit_configs = MoveItConfigs()
+    __robot_name: str
+    __urdf_package: Path
+    __urdf_filename: str
+    __srdf_filename: str
+    __robot_description: str
+
+    # Look-up for robot_name_moveit_config package
+    def __init__(self, robot_name: str, robot_description="robot_description"):
+        super().__init__(robot_name + "_moveit_config")
+        self.__robot_name = robot_name
+        setup_assistant_file = self._package_path / ".setup_assistant"
+        if not setup_assistant_file.exists():
+            logging.warning(
+                f"\x1b[33;21mPackage `{self._package_path}` doesn't have `.setup_assistant` file "
+                f"-- using config/{robot_name}.urdf and config/{robot_name}.srdf\x1b[0m"
+            )
+            self.__urdf_package = self._package_path
+            self.__urdf_filename = "config/" + self.__robot_name + ".urdf"
+            self.__srdf_filename = "config/" + self.__robot_name + ".srdf"
+        else:
+            setup_assistant_yaml = load_yaml(setup_assistant_file)
+            self.__urdf_package = Path(
+                get_package_share_directory(
+                    setup_assistant_yaml["moveit_setup_assistant_config"]["URDF"][
+                        "package"
+                    ]
+                )
+            )
+            self.__urdf_filename = setup_assistant_yaml[
+                "moveit_setup_assistant_config"
+            ]["URDF"]["relative_path"]
+            self.__srdf_filename = setup_assistant_yaml[
+                "moveit_setup_assistant_config"
+            ]["SRDF"]["relative_path"]
+        self.__robot_description = robot_description
+
+    def robot_description(self, file_path: Optional[str] = None, mappings: dict = None):
+        if file_path is None:
+            robot_description_file_path = self.__urdf_package / self.__urdf_filename
+        else:
+            robot_description_file_path = self._package_path / file_path
+        self.__moveit_configs.robot_description = {
+            self.__robot_description: load_xacro(
+                robot_description_file_path, mappings=mappings
+            )
+        }
+        return self
+
+    def robot_description_semantic(
+        self, file_path: Optional[str] = None, mappings: dict = None
+    ):
+        self.__moveit_configs.robot_description_semantic = {
+            self.__robot_description
+            + "_semantic": load_xacro(
+                self._package_path / (file_path or self.__srdf_filename),
+                mappings=mappings,
+            )
+        }
+        return self
+
+    def robot_description_kinematics(self, file_path: Optional[str] = None):
+        self.__moveit_configs.robot_description_kinematics = {
+            self.__robot_description
+            + "_kinematics": load_yaml(
+                self._package_path / (file_path or "config/kinematics.yaml")
```

Comment:
All file paths with explicit `/` in a string will fail on windows, won't they?
I guess you have to use a filepath class for these defaults too?